src/sentry/web/forms/accounts.py

```diff
@@ -337,7 +319,14 @@ def _clean_managed_field(self, field):
         return self.cleaned_data[field]
 
     def clean_email(self):
-        return self._clean_managed_field('email')
+        value = self._clean_managed_field('email').lower()
+        if User.objects.filter(email__iexact=value).exclude(id=self.user.id).exists()\
+                or User.objects.filter(username__iexact=value).exclude(id=self.user.id).exists():
```

Comment:
I think you could make this one query using a [Q object](https://docs.djangoproject.com/en/1.7/topics/db/queries/#complex-lookups-with-q-objects). Pretty sure this would do the same thing:

```python
from django.db.models import Q

User.objects.filter(Q(email__iexact=value) | Q(username__iexact=value)).exclude(id=self.user.id).exists()
```